\documentclass{article}

\begin{document}

\section{Introduction}
\subsection{Synopsis}
Package {\tt cgmisc} contains miscellaneous functions, hopefully useful for extending genome-wide association study (GWAS) analyses. 

\subsection{Getting help}
Like every other R function, the functions provided in this package are documented in the standard R-help (Rd) format and can be easily accessed by issuing {\tt help()} or its shorter version, {\tt ?} function. For instance, if you want to get more information on how to use the {\tt clump.markers()} function, type either {\tt help(clumpmarkers)} or {\tt ?clump.markers} and press return/enter. To see this document from within R you type {\tt vignette('cgmisc')}. 

\subsection{Purpose of this document}
This document aims at presenting how to use functions provided in this package in a typical GWAS data analyses workflow. It is, however, not pretending to be a GWAS tutorial as such.

\subsection{Conventions}
\begin{itemize}
  \item{All R commands are written in terminal type: {\tt myfun(foo=T, bar=54)}}
  \item{In the above example: {\tt myfun} is a \textit{function} and both {\tt foo} and {\tt bar} are its \textit{arguments}}
\end{itemize}

\section{Working with {\tt cgmisc}}
\subsection{Installation}
In order to install {\tt cgmisc}, you either use one of the R GUIs (native R GUI, RStudio etc.) or type the following command:
<<installation, tidy=TRUE, eval=FALSE>>=
install.packages("cgmisc", repos="")
@
\noindent Functions in the {\tt cgmisc} package often complement or use {\tt GenABEL} \ref{aulchenko} package functions and data structures. {\tt GenABEL} is an excellent and widely-used R package for performing genome-wide association studies and much more... Therefore {\tt GenABEL} will be loaded automagically when loading {\tt cgmisc}. If for some mysterious reason this does not happen, you can install and load {\tt GenABEL} by typing:
<<genabel, tidy=TRUE,eval=F>>=
install.packages('GenABEL')
require('GenABEL')
@
\noindent You load {\tt cgmisc} package in exactly the same way (both {\tt require} and {\tt library} will do):
<<loading, tidy=TRUE, eval=TRUE>>=
require('cgmisc')
library('cgmisc') # Alternative to require
@
\noindent After having loaded the package it is time to load some data:
<<loading.data, tidy=TRUE, eval=TRUE>>=
data(cgmiscdat1)
@
\section{Using {\tt plot.Manhattan.LD} function}
The {\tt plot.Manhattan.LD} function allows you to visualize the LD pattern in a genome fragment on an enchanced Manhattan plot. You select one marker, typically the one with the strongest association to the analysed trait and all other markers in the region are coloured according to the degree of linkage disequilibrium with this index marker. You need to begin by running a standard GWAS analyses:
<<plot.Manhattan.LD, tidy=TRUE, eval=TRUE, fig.show=TRUE, fig.ext='png', cache=TRUE>>=
# Run association analyses
an <- qtscore(cad~sex, data=cgmiscdat1) 
plot(an, pch=19, cex=.5) # Plot standard Manhattan 
@
Once this is done, you might be interested in checking the top associated marker. This can be done using the following call:
<<plot.Manhattan.LD.2, tidy=TRUE, eval=TRUE>>=
summary(an, top=5) # List top 5 markers
@
We can see that {\tt  ΒΙCF2P453669} on chromosome 6 (23.34Mb) is the strongest association. Now time to visualize a short fragment of chromosome 6, 1Mb downstream and 1Mb upstream of the top-associated marker.
<<plot.Manhattan.LD.3, tidy=TRUE, eval=TRUE>>=
plot.manhattan.LD(data=cgmiscdat1,gwas.result=an,chr=6,region=c(22340000, 24340000),index.snp="BICF2P453669",p.value=0.05,bonferroni=F,mafThreshold=.1)
@
\end{document}